src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -541,6 +541,67 @@ public function dumpFile($filename, $content, $mode = 0666)
     }
 
     /**
+     * Returns the "chmod" of the file or directory
+     *
+     * @param string $path A file or directory path.
+     * @param boolean $groups Get file|directory permissions by groups ex: 'u-wxrw--wt'.
+     *                        formats:
+     *                        s => socket;
+     *                        l => symbolic link;
+     *                        - => regular;
+     *                        b => block special;
+     *                        d => directory;
+     *                        c => character special;
+     *                        p => FIFO pipe;
+     *                        u => unknown.
+     *
+     * @return string Permissions of the file|directory or throw an exception on failure.
+     */
+    public function permissions($path, $groups = false)
+    {
+        if (!is_dir($path) && !is_file($path)) {
+            throw new IOException(sprintf('Unable to guess "%s" type.', $path), 0, null, $path);
+        }
+
+        $perms = substr(sprintf('%o', fileperms($path)), -4);
+
+        if (!$groups) {
+            return $perms;
+        }
+
+        if (($perms & 0xC000) == 0xC000) {
+            $out = 's';
+        } elseif (($perms & 0xA000) == 0xA000) {
+            $out = 'l';
+        } elseif (($perms & 0x8000) == 0x8000) {
+            $out = '-';
+        } elseif (($perms & 0x6000) == 0x6000) {
+            $out = 'b';
+        } elseif (($perms & 0x4000) == 0x4000) {
+            $out = 'd';
+        } elseif (($perms & 0x2000) == 0x2000) {
+            $out = 'c';
+        } elseif (($perms & 0x1000) == 0x1000) {
```

Comment:
Would it be an idea to move all those hexes to constants so everyone knows what they mean?